pkg/utils/util_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright © 2019 Doppler <support@doppler.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package utils
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+)
+
+var username string
+var homeDir string
+var cwd string
+
+func init() {
+	currentUser, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
+	username = currentUser.Username
+
+	homeDir, err = os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+
+	cwd, err = filepath.Abs(".")
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestParsePathTilde(t *testing.T) {
+	path, err := ParsePath("~")
+	if err != nil || path != homeDir {
+		t.Error(fmt.Sprintf("Got %s, expected %s", path, homeDir))
+	}
+
+	path, err = ParsePath("~/")
+	if err != nil || path != homeDir {
+		t.Error(fmt.Sprintf("Got %s, expected %s", path, homeDir))
+	}
+
+	path, err = ParsePath(fmt.Sprintf("~%s", username))
+	if err != nil || path != homeDir {
+		t.Error(fmt.Sprintf("Got %s, expected %s", path, homeDir))
+	}
+
+	path, err = ParsePath(fmt.Sprintf("~%s/", username))
+	if err != nil || path != homeDir {
+		t.Error(fmt.Sprintf("Got %s, expected %s", path, homeDir))
+	}
+
+	// expect error
+	path, err = ParsePath(fmt.Sprintf("~%s/", username+"1"))
+	if err == nil || path != "" {
+		t.Error(fmt.Sprintf("Got %s, expected error", path))
+	}
+}
+
```

Comment:
More tests:

- `~/~`
- `~user/~otheruser`
- `~otheruser/~user`
- `~user/content/to/path/~`

Reply:
This function is meant as a helper to parse a leading tilde, rather than as a strict validator against invalid syntax.